src/store/index.ts

```diff
@@ -3,21 +3,27 @@ import { composeWithDevTools } from 'redux-devtools-extension'
 import { connectRouter, routerMiddleware } from 'connected-react-router'
 import thunk from 'redux-thunk'
 
-import globalReducer from './global/global.reducer'
+import globalReducer, { GlobalState } from './global/global.reducer'
 import homeReducer from './home/home.reducer'
 import myAccountReducer from './my-account/my-account.reducer'
-import accountDetailsReducer from './account-details/account-details.reducer'
+import accountDetailsReducer, { AccountDetailsState } from './account-details/account-details.reducer'
 import transactionDetailsReducer from './transaction-details/transaction-details.reducer'
 import transactionReducer from './transaction/transaction.reducer'
 import loginReducer from './login/login.reducer'
 import tokenSwapReducer from './token-swap/token-swap.reducer'
+import { History } from 'history'
+
+export interface RootState {
+  accountDetails: AccountDetailsState;
+  global: GlobalState;
+}
 
 /**
  * Creates the Redux store root reducer combining all the reducers used in the app
  * @param {History} history - Browser history
  * @returns {Object} - Root reducer
  */
-function createRootReducer (history) {
+export function createRootReducer (history: History<RootState>) {
```

Comment:
Btw, I've just seen this `history: History<RootState>`. I think that this is not correct, as the history is the browser history and should not have anything to do wit the state of the reducer

Reply:
Damn you're right I messed up there! Good catch! Fixed!